### PR TITLE
Add interactive map

### PR DESCRIPTION
**Pull request by tinykite**
## Summary 
Adds an interactive map that is accessible by hovering, tabbing, or using the keyboard. The map is navigable by arrow keys if you tab onto the map and then hit the left, right, down, or up arrow. 

## Video Preview

https://github.com/harvard-lil/capstone-static/assets/4039311/e110660d-5bb7-46a0-905d-3fffd9510053

## What this does
- Adds a reusable map component, `accessible-map`
- Updates the previous placeholder `caselaw-map` component with an interactive map
- Adds a hashmap, `map.js` that provides information on each state in the map: path data, adjacent neighbors, and display name
- Adds an array of map abbreviations for easily outputting an svg group for each state in the map
- Reformats files with Prettier because I discovered I hadn't enabled formatting on save 😮‍💨 

## Notes 
- The current case.law map features a skip link to allow users to skip over the map. This will be added when I build out the CAP navigation components. 
- I will also be adding an `aria-live` region to the sidebar when the reporter stats update based on JSON data
- Arc/Chrome by default treats links as focusable elements — Firefox and Safari do not. Which means that by default, the map cannot be navigable by tab only. This is somewhat confusing behavior, but (apparently) intended — you can read more about [what Scott O'Hara has written about tabbing links](https://www.scottohara.me/blog/2014/10/03/link-tabbing-firefox-osx.html), as well as  some more up-to-date info from the [A11Y Project. ](https://www.a11yproject.com/posts/macos-browser-keyboard-navigation/)
- The keyboard navigation pattern I adopted is similar to [this example of Visa Chart Components](https://observablehq.com/@frankelavsky/chart-component-boilerplate). Notably, it surfaces a tooltip that alerts users how they can navigate the chart. Adding a small, but noticeable tooltip icon to CAP's homepage is also on my follow-up list of work for the homepage.

### CODE REVIEW

File: components/accessible-map.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+					d="M222.989 9.4405L213.836 34.5876M222.989 9.4405C220.96 8.43697 219.967 6.04644 220.76 3.86917C221.61 1.53403 224.192 0.329727 226.528 1.17964C228.863 2.02956 230.067 4.61222 229.217 6.94735C228.367 9.28343 225.785 10.4868 223.449 9.63688C223.291 9.57908 223.137 9.5135 222.989 9.4405Z"
+					stroke="white"
+					stroke-width="0.5"
+				/>
+
+				${this.abbreviations.map(
+					(abbreviation) =>
+						svg`<g class="map__group">
+				<title id=${abbreviation}>${mapData[abbreviation].name}</title>
+				<a data-abbreviation=${abbreviation} @focus="${this._handlePointerEvent}"  href="https://tinykitelab.com" class="map__link map__link--${abbreviation}" aria-labelledby=${abbreviation}>
```
Comment (tinykite) [inline components/accessible-map.js:106]: Including a title + a link here results in somewhat verbose output in Voiceover, (A state will get read aloud twice) but I'm not sure removing the title would be a better option due to the fact that the titles are often presented as tooltips by browsers. 

Opted to include `aria-labelledby` based on Adrian Roselli's article [My Priority of Methods for Labeling a Control](https://adrianroselli.com/2020/01/my-priority-of-methods-for-labeling-a-control.html). I generally avoid using `aria-label` if possible based on Adrian's advice as well as Eric Bailey's article [aria-label is a code smell](https://ericwbailey.website/published/aria-label-is-a-code-smell/)

Comment (tinykite) [reply components/accessible-map.js:102]: Tiny additional note: I will definitely be changing the link to the correct path when the pages exist!

File: components/caselaw-map.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+						<p class="infoBox__text">
+							6,930,777<br /><span class="infoBox__textDescriptor"
+								>Unique cases</span
+							>
+						</p>
+						<p></p>
+						<p class="infoBox__text">
+							612<br /><span class="infoBox__textDescriptor">Reporters</span>
+						</p>
+						<p></p>
```
Comment (rebeccacremona) [inline components/caselaw-map.js:120]: The tiniest of questions: are these empty `<p></p>` intentional, or just relics of coding? If they are intentional, I am not opining that they be removed 🙂 .

**Review by rebeccacremona — APPROVED** [review]
This is wicked cool!

I love how clean the markup is. I keep being impressed by what can be achieved with custom elements like this, and am jazzed to see that playing out here.

File: components/caselaw-map.js (hunk 2 above)
Comment (tinykite) [reply components/caselaw-map.js:120]: Oh no, haha they are not intentional and were totally missed in my code review! Will remove them, thank you for catching!

**Comment by tinykite** [comment]
Thanks @rebeccacremona!

**Pull request merged**